### PR TITLE
[#DEV-7341] Handles Better Webgl Context Lost

### PR DESCRIPTION
**Pull request by Kuranes**
Once context Lost, prevent any further render, inform osgjs client code with a callback
Otherwise client may continue to ask new renders and thus getting
endless errors.

### CODE REVIEW

**Comment by Kuranes** [comment]
add unit test using webgl debug js context lost simulator

File: sources/osgViewer/Viewer.js [hunk 1]
```diff
@@ -193,8 +193,15 @@ Viewer.prototype = MACROUTILS.objectInherit( View.prototype, {
...
 
+    setContextLostCallback: function ( cb ) {
+        this._contextLostCallback = cb;
+    },
     contextLost: function () {
         Notify.log( 'webgl context lost' );
+        if ( this._contextLostCallback ) {
+            this._contextLostCallback();
+        }
+        this._contextLost = true;
```
Comment (cedricpinson) [inline sources/osgViewer/Viewer.js]: this._contextLost is initialized somewhere ?

Comment (Kuranes) [inline sources/osgViewer/Viewer.js]: no, undefined by defaut.

Comment (cedricpinson) [inline sources/osgViewer/Viewer.js]: It's better to initialize vars in the constructor, it's easier to know member variables used in the class. The few exceptions is when we need to call a init method that duplicate a lot of code initialized.

**Pull request merged**